### PR TITLE
New package: yuk7.ArchWSL version 24.4.28.0

### PR DESCRIPTION
**Pull request by schembriaiden**
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/157014)

### CODE REVIEW

**Comment by stephengillie** [comment]
Automatic Validation ended with:


(Automated response - build 884.)

**Review by stephengillie — CHANGES_REQUESTED** [review]
Hi @vikingnope,

Manual Validation ended with:
image

I'm not sure why the application isn't picking up the dependency, but I believe this dependency has another way to be installed. 

Edit: The error code is WSL-specific:
```PowerShell
PS C:\ManVal> winget error 0x8007019E
0x8007019e
The Windows Subsystem for Linux has not been enabled.
```

File: manifests/y/yuk7/ArchWSL/24.4.28.0/yuk7.ArchWSL.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+InstallerType: zip
+NestedInstallerType: exe
+NestedInstallerFiles:
+- RelativeFilePath: Arch.exe
+InstallModes:
+- silent
+- silentWithProgress
+Dependencies:
+  PackageDependencies:
+  - PackageIdentifier: Microsoft.WSL
```
Comment (stephengillie) [inline manifests/y/yuk7/ArchWSL/24.4.28.0/yuk7.ArchWSL.installer.yaml:15]:
```suggestion
  WindowsFeatures:
  - Microsoft-Windows-Subsystem-Linux
```
I forgot to hit "Start a revew".

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 
> 2024-06-11 17:06:55.964 [CORE] Enabling Windows Feature [Microsoft-Windows-Subsystem-Linux] 
> 2024-06-11 17:06:55.964 [CLI ] Starting: 'C:\Windows\System32\Dism.exe' with arguments '/Online /Enable-Feature /NoRestart /FeatureName:Microsoft-Windows-Subsystem-Linux' 
> 2024-06-11 17:07:21.836 [CORE] Failed to enable Windows Feature [Microsoft-Windows-Subsystem-Linux] with exit code: 14081 
> 2024-06-11 17:07:21.837 [CLI ] Terminating context: 0x8a150110 at C:\__w\1\s\external\pkg\src\AppInstallerCLICore\Workflows\DependenciesFlow.cpp:d4  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean  

Automated error analysis suggests  may mean 

(Automated response - build 884.)

**Comment by stephengillie** [comment]
Apologies - we're working through some ephemeral capacity issues for the validation pipeilne.

**Comment by vikingnope** [comment]
No problem, just wanted to make sure it wasn't an issue from my end

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 12:02:37 PM: 
> 2024-07-15 12:02:37.326 [CORE] Failed to enable Windows Feature [Microsoft-Windows-Subsystem-Linux] with exit code: 14081

Automated error analysis suggests 14081 may mean The referenced assembly could not be found. Is a dependency missing?

(Automated response - build 886.)

**Comment by vikingnope** [comment]
> Manual Validation ended with:
> 
> > 12:02:37 PM:
> > 2024-07-15 12:02:37.326 [CORE] Failed to enable Windows Feature [Microsoft-Windows-Subsystem-Linux] with exit code: 14081
> 
> Automated error analysis suggests 14081 may mean The referenced assembly could not be found. Is a dependency missing?
> 
> (Automated response - build 886.)

Doesn't seem to be

image

**Comment by vikingnope** [comment]
> This might be due to both the validation pipeline VMs, and my VMs, both having an issue enabling WSL.

Ah yes I think so

**Comment by vikingnope** [comment]
Are we able to merge this @stephengillie?

**Comment by stephengillie** [comment]
> Are we able to merge this @stephengillie?

I don't have enough information at this time to give a valid answer. I do not know when this situation will change.

**Comment by stephengillie** [comment]
Manually enabling WSL gives:

**Comment by vikingnope** [comment]
> Manually enabling WSL gives: image


Is this on a VM or an actual windows build?

**Comment by stephengillie** [comment]
> Is this on a VM or an actual windows build?

VM downloaded from Hyper-V Quick Create. Unforunately, we don't currently have any bare metal testing systems.

**Comment by vikingnope** [comment]
> > Is this on a VM or an actual windows build?
> 
> VM downloaded from Hyper-V Quick Create. Unforunately, we don't currently have any bare metal testing systems.

I am 99.9% sure that that is the issue because I am pretty sure you cannot have WSL running on a hypervisor since there is no pass through of virtualization from the CPU directly on hypervisors. (or at least that how I think it works), so for now maybe mark this as blocked?

**Comment by ItzLevvie** [comment]
[Policy] Blocking-Issue

**Pull request closed without merging**